src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2184,14 +2184,22 @@ private static bool HasInstanceData(MemberDeclarationSyntax m)
                         !ContainsModifier(fieldDecl.Modifiers, SyntaxKind.StaticKeyword) &&
                         !ContainsModifier(fieldDecl.Modifiers, SyntaxKind.ConstKeyword);
                 case SyntaxKind.PropertyDeclaration:
-                    // auto-property
-                    var propertyDecl = (PropertyDeclarationSyntax)m;
-                    return
-                        !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.StaticKeyword) &&
-                        !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.AbstractKeyword) &&
-                        !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.ExternKeyword) &&
-                        propertyDecl.AccessorList != null &&
-                        All(propertyDecl.AccessorList.Accessors, a => a.Body == null && a.ExpressionBody == null);
+                    {
+                        // auto-property
+                        var propertyDecl = (PropertyDeclarationSyntax)m;
+                        return
+                            !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.StaticKeyword) &&
+                            !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.AbstractKeyword) &&
+                            !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.ExternKeyword) &&
+                            propertyDecl.AccessorList != null &&
+                            All(propertyDecl.AccessorList.Accessors, a => a.Body == null && a.ExpressionBody == null);
+                    }
+                case SyntaxKind.DataPropertyDeclaration:
+                    {
+                        // auto-property
+                        var propertyDecl = (DataPropertyDeclarationSyntax)m;
+                        return !ContainsModifier(propertyDecl.Modifiers, SyntaxKind.AbstractKeyword);
```

Comment:
>!ContainsModifier(propertyDecl.Modifiers, SyntaxKind.AbstractKeyword) [](start = 31, length = 69)

Would it be more robust long term to share the modifiers check with the regular properties? Even if some modifiers would be an error on this declaration.